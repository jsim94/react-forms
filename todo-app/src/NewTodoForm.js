import { useState } from "react";

function NewTodoForm({ submit, init }) {
  const initData = init || {
    content: "",
  };
  const [formData, setFormData] = useState(initData);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    submit(formData);
    setFormData(initData);
  };

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="content">Todo</label>
      <input
        id="content"
        name="content"
        value={formData.content}
        type="text"
        onChange={handleChange}
      />
      <input type="submit" />
    </form>
  );
}

export default NewTodoForm;
