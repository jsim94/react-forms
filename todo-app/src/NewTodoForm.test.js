import { render, fireEvent } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("should render", function () {
  render(<NewTodoForm />);
});

it("should match snapshot", function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
