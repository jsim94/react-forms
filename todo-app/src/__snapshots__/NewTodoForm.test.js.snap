// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="content"
    >
      Todo
    </label>
    <input
      id="content"
      name="content"
      type="text"
      value=""
    />
    <input
      type="submit"
    />
  </form>
</DocumentFragment>
`;
